src/defaults/render-actions.js

```diff
@@ -69,13 +73,21 @@ const Actions = ({
       )
     }
 
-    if (isRecording) {
-      return <StopButton onClick={onStopRecording} data-qa='stop-recording' />
-    }
+    const { elapsedSeconds } = this.state
+    const maxSeconds = Math.floor(timeLimit / 1000)
+    const maxReadySeconds = Math.floor(countdownTime / 1000)
 
-    if (isCameraOn && streamIsReady) {
+    if (isRecording || (isCameraOn && streamIsReady)) {
       return (
-        <RecordButton onClick={onStartRecording} data-qa='start-recording' />
+        <PrimaryButtonComponent
+          onClick={isRecording ? onStopRecording : onStartRecording}
+          maxSeconds={maxSeconds}
+          status={isRecording ? 'recording' : ''}
+          data-qa={isRecording ? 'stop-recording' : 'start-recording'}
+          elapsedSeconds={elapsedSeconds}
+          readySeconds={0}
+          maxReadySeconds={maxReadySeconds}
+        />
```

Comment:
This is definitely becoming VERY specific to the needs and I don't see it merging back to the base library.